server/controllers/api/videos/ownership.ts

```diff
@@ -0,0 +1,108 @@
+import { VideoChangeOwnershipCreate } from '../../../../shared/models/videos'
+import { VideoChannelModel } from '../../../models/video/video-channel'
+import { getFormattedObjects } from '../../../helpers/utils'
+
+const ownershipVideoRouter = express.Router()
+
+ownershipVideoRouter.post('/:id/give-ownership',
+  authenticate,
+  asyncMiddleware(videosChangeOwnershipValidator),
+  asyncRetryTransactionMiddleware(giveVideoOwnership)
+)
+
+ownershipVideoRouter.get('/ownership',
+  authenticate,
+  asyncRetryTransactionMiddleware(listVideoOwnership)
+)
+
+ownershipVideoRouter.post('/ownership/:id/accept',
+  authenticate,
+  asyncMiddleware(videosTerminateChangeOwnershipValidator),
+  asyncMiddleware(videosAcceptChangeOwnershipValidator),
+  asyncRetryTransactionMiddleware(acceptOwnership)
+)
+
+ownershipVideoRouter.post('/ownership/:id/refuse',
+  authenticate,
+  asyncMiddleware(videosTerminateChangeOwnershipValidator),
+  asyncRetryTransactionMiddleware(refuseOwnership)
+)
+
+// ---------------------------------------------------------------------------
+
+export {
+  ownershipVideoRouter
+}
+
+// ---------------------------------------------------------------------------
+
+async function giveVideoOwnership (req: express.Request, res: express.Response) {
+  const videoInstance = res.locals.video as VideoModel
+  const initiatorAccount = res.locals.oauth.token.User.Account as AccountModel
+  const body: VideoChangeOwnershipCreate = req.body
+
+  await sequelizeTypescript.transaction(async t => {
+    const nextOwner = await AccountModel.findOne({ where: { name: body.username } })
+    if (nextOwner) {
+      const videoChangeOwnershipToCreate = {
+        initiatorAccountId: initiatorAccount.id,
+        nextOwnerAccountId: nextOwner.id,
+        videoId: videoInstance.id
+      }
+      await VideoChangeOwnershipModel.create(videoChangeOwnershipToCreate, { transaction: t })
+
+      logger.info('Ownership change for video %s created.', videoInstance.name)
+      return res.type('json').status(204).end()
+    } else {
+      return res.type('json').status(400).end()
+    }
+  })
+}
+
+async function listVideoOwnership (req: express.Request, res: express.Response) {
+  const currentAccount = res.locals.oauth.token.User.Account as AccountModel
+  const resultList = await VideoChangeOwnershipModel.listForApi(
+    currentAccount.id,
+    req.query.start || 0,
+    req.query.count || 10,
+    req.query.sort || 'createdAt'
+  )
+
+  return res.json(getFormattedObjects(resultList.data, resultList.total))
+}
+
+async function acceptOwnership (req: express.Request, res: express.Response) {
+  return sequelizeTypescript.transaction(async t => {
+    const videoChangeOwnership = res.locals.videoChangeOwnership as VideoChangeOwnershipModel
+    const targetVideo = videoChangeOwnership.Video
+    const channel = res.locals.videoChannel as VideoChannelModel
+
+    targetVideo.set('channelId', channel.id)
```

Comment:
Now comes the ActivityPub parts. 
 * Send video update. This might not work for now because in the `processUpdate` function we check the actor owns the video channel. I think we should send the activity from the old actor (so the activity is signed with the old owner) and in the `processUpdate` function, check the `attributedTo` `Person` owns the video channel (instead of the `videoChannel.Account.Actor.id !== actor.id` condition). Send me en email if you need more help
 * Use `changeVideoChannelShare` method